metadata-integration/java/examples/src/main/java/io/datahubproject/examples/DatasetCustomPropertiesAddRemove.java

```diff
@@ -0,0 +1,61 @@
+package io.datahubproject.examples;
+
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.mxe.MetadataChangeProposal;
+import datahub.client.MetadataWriteResponse;
+import datahub.client.patch.PatchOperationType;
+import datahub.client.patch.dataset.DatasetPropertiesPatchBuilder;
+import datahub.client.rest.RestEmitter;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+class DatasetCustomPropertiesAddRemove {
+
+  private DatasetCustomPropertiesAddRemove() {
+
+  }
+
+  /**
+   * Applies Add and Remove property operations on an existing custom properties aspect without
+   * affecting any other properties
+   * @param args
+   * @throws IOException
+   * @throws ExecutionException
+   * @throws InterruptedException
+   */
+    public static void main(String[] args) throws IOException, ExecutionException, InterruptedException {
+      MetadataChangeProposal datasetPropertiesProposal = new DatasetPropertiesPatchBuilder()
+          .urn(UrnUtils.toDatasetUrn("hive", "fct_users_deleted", "PROD"))
+          // the operation type doesn't matter here, what matters are addProperty / removeProperty calls below
+          .op(PatchOperationType.ADD)
```

Comment:
@RyanHolstien : looking at this now, the `op` probably doesn't make sense on the top level builder ever..   we should have `add_x`, `remove_x` or `set_x` methods which imply the `op` on the field called `x`. See the Python sdk example.

Reply:
The idea here was to limit what was possible to make the interface simpler. If you want to do a different operation you make a new proposal rather than doing adds, replaces, and deletes within the same one as it's really easy to shoot yourself in the foot and create an invalid patch. It also makes it so you don't need a patch builder style interface per every field and instead it's at the aspect level.

I think it's a bit more intuitive and usable this way without having to read the full JsonPatch docs since this is targeted at people who wouldn't want to construct one themselves since that is also an option on the raw ingestProposal endpoint, but I'm open to rethinking the interface if we think it's too restrictive.